src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -108,66 +109,148 @@ def get_active_recovery_threshold():
-            return True
-
-        # Run region checks and updates once an hour
-        runs_per_hour = UptimeSubscription.IntervalSeconds.ONE_HOUR / subscription.interval_seconds
-        subscription_run = UUID(subscription.subscription_id).int % runs_per_hour
-        current_run = (
-            datetime.fromtimestamp(result["scheduled_check_time_ms"] / 1000, timezone.utc).minute
-            * 60
-        ) // subscription.interval_seconds
-        if subscription_run == current_run:
-            return True
 
+def should_run_region_checks(subscription: UptimeSubscription, result: CheckResult) -> bool:
+    if not subscription.subscription_id:
+        # Edge case where we can have no subscription_id here
         return False
 
-    def check_and_update_regions(
-        self,
-        subscription: UptimeSubscription,
-        result: CheckResult,
-        regions: list[UptimeSubscriptionRegion],
-    ):
-        """
-        This method will check if regions have been added or removed from our region configuration,
-        and updates regions associated with this uptime monitor to reflect the new state. This is
-        done probabilistically, so that the check is performed roughly once an hour for each uptime
-        monitor.
-        """
-        if not self.should_run_region_checks(subscription, result):
-            return
+    # XXX: Randomly check for updates once an hour - this is a hack to fix a bug where we're seeing some checks
+    # not update correctly.
+    chance_to_run = subscription.interval_seconds / timedelta(hours=1).total_seconds()
+    if random.random() < chance_to_run:
+        return True
+
+    # Run region checks and updates once an hour
+    runs_per_hour = UptimeSubscription.IntervalSeconds.ONE_HOUR / subscription.interval_seconds
+    subscription_run = UUID(subscription.subscription_id).int % runs_per_hour
+    current_run = (
+        datetime.fromtimestamp(result["scheduled_check_time_ms"] / 1000, timezone.utc).minute * 60
+    ) // subscription.interval_seconds
+    if subscription_run == current_run:
+        return True
+
+    return False
+
+
+def has_reached_status_threshold(
+    project_subscription: ProjectUptimeSubscription,
+    status: str,
+    metric_tags: dict[str, str],
+) -> bool:
+    pipeline = _get_cluster().pipeline()
+    key = build_active_consecutive_status_key(project_subscription, status)
+    pipeline.incr(key)
+    pipeline.expire(key, ACTIVE_THRESHOLD_REDIS_TTL)
+    status_count = int(pipeline.execute()[0])
+    result = (status == CHECKSTATUS_FAILURE and status_count >= get_active_failure_threshold()) or (
+        status == CHECKSTATUS_SUCCESS and status_count >= get_active_recovery_threshold()
+    )
+    if not result:
+        metrics.incr(
+            "uptime.result_processor.active.under_threshold",
+            sample_rate=1.0,
+            tags=metric_tags,
+        )
+    return result
+
+
+def try_check_and_update_regions(
```

Comment:
I've been thinking that we should move these checks to a regular task maybe, so this makes that a little easier